src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -1272,6 +1272,21 @@ export class EditorPart extends Part implements IEditorPart, IEditorGroupService
 			return Position.ONE; // can only be ONE
 		}
 
+		const config = this.configurationService.getConfiguration<IWorkbenchEditorConfiguration>();
+		const reuseIfOpen = config.workbench.editor.reuseIfOpen;
+		// Respect option to reveal an editor if it is open (not necessarily already visible)
+		const skipReuse = (options && options.index) || arg1;
+		if(!skipReuse && reuseIfOpen) {
+			const groups = this.stacks.groups;
```

Comment:
I suggest to extract this into a method `findGroup(editor)` in the `EditorStacksModel` and add tests for it. I would add an option to this method to only respect the active editor and then also use it for the `revealIfVisible` option further below 👍